# 캘리포니아 집 값 예측(회귀)

## 데이터 불러오기

In [27]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

from plotly.subplots import make_subplots

In [7]:
df = pd.read_csv("./data/housing.csv")

In [14]:
df.shape

(20640, 10)

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 10 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   longitude           20640 non-null  float64
 1   latitude            20640 non-null  float64
 2   housing_median_age  20640 non-null  float64
 3   total_rooms         20640 non-null  float64
 4   total_bedrooms      20433 non-null  float64
 5   population          20640 non-null  float64
 6   households          20640 non-null  float64
 7   median_income       20640 non-null  float64
 8   median_house_value  20640 non-null  float64
 9   ocean_proximity     20640 non-null  object 
dtypes: float64(9), object(1)
memory usage: 1.6+ MB


In [18]:
df.describe()

longitude      latitude  housing_median_age   total_rooms   
count  20640.000000  20640.000000        20640.000000  20640.000000  \
mean    -119.569704     35.631861           28.639486   2635.763081   
std        2.003532      2.135952           12.585558   2181.615252   
min     -124.350000     32.540000            1.000000      2.000000   
25%     -121.800000     33.930000           18.000000   1447.750000   
50%     -118.490000     34.260000           29.000000   2127.000000   
75%     -118.010000     37.710000           37.000000   3148.000000   
max     -114.310000     41.950000           52.000000  39320.000000   

       total_bedrooms    population    households  median_income   
count    20433.000000  20640.000000  20640.000000   20640.000000  \
mean       537.870553   1425.476744    499.539680       3.870671   
std        421.385070   1132.462122    382.329753       1.899822   
min          1.000000      3.000000      1.000000       0.499900   
25%        296.000000    787.000000    280.000000       2.563400   
50%        435.000000   1166.000000    409.000000       3.534800   
75%        647.000000   1725.000000    605.000000       4.743250   
max       6445.000000  35682.000000   6082.000000      15.000100   

       median_house_value  
count        20640.000000  
mean        206855.816909  
std         115395.615874  
min          14999.000000  
25%         119600.000000  
50%         179700.000000  
75%         264725.000000  
max         500001.000000

In [20]:
df.isna().sum()

longitude               0
latitude                0
housing_median_age      0
total_rooms             0
total_bedrooms        207
population              0
households              0
median_income           0
median_house_value      0
ocean_proximity         0
dtype: int64

## 데이터 확인
- 데이터 전체를 출력하는 이미지가 필요한데?
- 1) 3X3 레이아웃
- 2) 막대그래프를 어떻게든 생성!!
- longitude               0
- latitude                0
- housing_median_age      0
- total_rooms             0
- total_bedrooms        207
- population              0
- households              0
- median_income           0
- median_house_value      0
- ocean_proximity         0
- dtype: int64


In [26]:
!pip install plotly

## 데이터 전처리

In [40]:
fig = make_subplots(rows=3, cols=3, subplot_titles=("<i>longitude",
                                                    "<i>latitude",
                                                    "<i>housing_median_age",
                                                    "<i>total_rooms",
                                                    "<i>total_bedrooms",
                                                    "<i>population",
                                                    "<i>households",
                                                    "<i>median_income",
                                                    "<i>median_house_value",))
fig.add_traces(go.Histogram(x=df["longitude"], name="longitude"), row=1, col=1)
fig.add_traces(go.Histogram(x=df["latitude"], name="latitude"), row=1, col=2)
fig.add_traces(go.Histogram(x=df["housing_median_age"], name="housing_median_age"), row=1, col=3)
fig.add_traces(go.Histogram(x=df["total_rooms"], name="total_rooms"), row=1, col=1)
fig.add_traces(go.Histogram(x=df["population"], name="population"), row=1, col=1)
fig.add_traces(go.Histogram(x=df["households"], name="households"), row=1, col=1)
fig.add_traces(go.Histogram(x=df["median_income"], name="median_income"), row=1, col=1)
fig.add_traces(go.Histogram(x=df["median_house_value"], name="median_house_value"), row=1, col=1)
fig.update_layout(template="plotly_dark", title_text="<b>Distrubutions")

TypeError: Figure.add_traces() got an unexpected keyword argument 'row'

## 데이터 분리(머신러닝을 위해서)

## 데이터 학습 및 검증